File: infrastructure/api-gateway/src/main/java/com/example/chat/gateway/ApiGatewayApplication.java
package com.example.chat.gateway;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.cloud.client.discovery.EnableDiscoveryClient;

import com.example.chat.common.logging.annotation.EnableTracingLogging;

/**
 * API Gateway Application
 *
 * 담당 기능:
 * - 라우팅 (서비스 디스커버리 기반 lb://)
 * - 로깅
 * - 인증
 * - Rate Limiting
 */
@EnableTracingLogging
@EnableDiscoveryClient
@SpringBootApplication
public class ApiGatewayApplication {

    public static void main(String[] args) {
        SpringApplication.run(ApiGatewayApplication.class, args);
    }
}

File: infrastructure/api-gateway/src/main/kotlin/com/example/chat/gateway/ApiGatewayApplication.kt
package com.example.chat.gateway

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.cloud.client.discovery.EnableDiscoveryClient

/**
 * Spring Cloud API Gateway
 *
 * 모든 마이크로서비스의 단일 진입점
 * - 라우팅
 * - 로드 밸런싱
 * - 인증/인가
 * - Rate Limiting
 */
@EnableDiscoveryClient
@SpringBootApplication
class ApiGatewayApplication

fun main(args: Array<String>) {
	runApplication<ApiGatewayApplication>(*args)
}
